.github/scripts/generate_binary_build_matrix.py

```diff
@@ -57,58 +80,89 @@
+    return [
+        {
+            "python_version": python_version,
+            "gpu_arch_type": arch_type(arch_version),
+            "gpu_arch_version": arch_version,
+            "container_image": CONDA_CONTAINER_IMAGES[arch_version],
+        }
+        for python_version in snip_if(is_pr, FULL_PYTHON_VERSIONS)
+        # We don't currently build conda packages for rocm
+        for arch_version in ["cpu"] + snip_if(is_pr, CUDA_ARCHES)
+    ]
+
+
+def generate_libtorch_matrix(is_pr: bool) -> List[Dict[str, str]]:
+    libtorch_variants = [
+        "shared-with-deps",
+        "shared-without-deps",
+        "static-with-deps",
+        "static-without-deps",
+    ]
+    return [
+        {
+            "gpu_arch_type": arch_type(arch_version),
+            "gpu_arch_version": arch_version,
+            "libtorch_variant": libtorch_variant,
+            "devtoolset": abi_version,
+            "container_image": LIBTORCH_CONTAINER_IMAGES[(arch_version, abi_version)],
+        }
+        # We don't currently build libtorch for rocm
+        for arch_version in ["cpu"] + snip_if(is_pr, CUDA_ARCHES)
+        for libtorch_variant in libtorch_variants
+        # one of the values in the following list must be exactly
+        # "cxx11-abi", but the precise value of the other one doesn't
+        # matter
+        for abi_version in ["cxx11-abi", "pre-cxx11"]
+    ]
+
+
+def generate_wheels_matrix(is_pr: bool) -> List[Dict[str, str]]:
     arches = ["cpu"]
-    if is_pull_request():
-        python_versions = [python_versions[-1]]
-        cuda_versions = [cuda_versions[-1]]
-        rocm_versions = [rocm_versions[-1]]
-    if include_cuda:
-        arches += cuda_versions
-    if include_rocm:
-        arches += rocm_versions
-    matrix = []
-    for item in itertools.product(python_versions, arches):
-        python_version, arch_version = item
-        # Not my favorite code here
-        gpu_arch_type = "cuda"
-        if "rocm" in WHEEL_CONTAINER_IMAGES[arch_version]:
-            gpu_arch_type = "rocm"
-        elif "cpu" in WHEEL_CONTAINER_IMAGES[arch_version]:
-            gpu_arch_type = "cpu"
-        matrix.append({
+    arches += snip_if(is_pr, CUDA_ARCHES)
+    arches += snip_if(is_pr, ROCM_ARCHES)
+    return [
+        {
             "python_version": python_version,
-            "gpu_arch_type": gpu_arch_type,
+            "gpu_arch_type": arch_type(arch_version),
             "gpu_arch_version": arch_version,
-            "wheel_container_image": WHEEL_CONTAINER_IMAGES.get(
-                arch_version, ""
-            ),
-            "conda_container_image": CONDA_CONTAINER_IMAGES.get(
-                arch_version, ""
-            )
-        })
-    return json.dumps({"include": matrix})
-
-def main():
-    include_cuda = True
-    include_rocm = True
-    if os.environ.get("NO_ROCM", False):
-        include_rocm = False
-    if os.environ.get("NO_CUDA", False):
```

Comment:
are we still going to support these?

Reply:
these env vars? no; they're encoded in the various `generate_*_matrix` functions